### PR TITLE
#160921881 User signup feature

### PR DESCRIPTION
**Pull request by obulaworld**
#### What does this PR do?
-It completes the user signup api feature.

#### Description of Task to be completed?
- Complete the user initialize registration feature
- Complete the user activation feature
- Complete the user update feature

#### How should this be manually tested?
- Clone the repo
- cd into the 'lotus-ah'
- Run 'npm install'
- Use the .env.sample format to create a .env file
- Run 'npm run migrate'
- Run 'npm run start:dev'

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[#160921881](https://www.pivotaltracker.com/story/show/160921881")

####Questions
- NA

### CODE REVIEW

**Review by Mrshinezee — APPROVED** [review]
LGTM

**Review by Victor-Ugwueze — APPROVED** [review]
LGTM

File: server/db/models/user.js [hunk 1]
```diff
@@ -26,27 +29,35 @@ const user = (sequelize, DataTypes) => {
...
     },
     favouriteTags: {
-      type: DataTypes.ARRAY(DataTypes.STRING)
-    },
-    roleId: {
-      type: DataTypes.INTEGER
+      type: DataTypes.ARRAY(DataTypes.STRING),
+      allowNull: true
     },
+    // roleId: {
```
Comment (KaiserPhemi) [inline server/db/models/user.js]: #### Nitpick
if these lines of code are not needed please remove them

**Review by KaiserPhemi — CHANGES_REQUESTED** [review]

File: server/db/models/user.js (hunk 1 above)
Comment (obulaworld) [reply server/db/models/user.js]: Okay. okay. on it

**Review by KaiserPhemi — APPROVED** [review]

**Pull request merged**